src/components/routes/homepage/Events.js

```diff
@@ -14,17 +19,41 @@ const useStyles = makeStyles(theme => ({
     [theme.breakpoints.down('xs')]: {
       gridTemplateColumns: 'repeat(1, 1fr)'
     }
+  },
+  noEvents: {
+    color: theme.palette.error.light,
+    padding: theme.spacing(2)
   }
 }));
 
 export const Events = ({ events }) => {
+  const [filter, setFilter] = useState('');
+  const [sortParam, setSortParam] = useState('');
+
   const classes = useStyles();
 
+  const filteredEvents = filterEvents({ events, filter, sortParam });
+
   return (
     <Container className={classes.container}>
-      {events.map(event => (
-        <MediaCard key={event.id} event={event} />
-      ))}
+      <SearchBar
+        setFilter={setFilter}
+        sortParam={sortParam}
+        setSortParam={setSortParam}
+      />
+      {filteredEvents.length > 0 ? (
+        <Container className={classes.events}>
+          {filterEvents({ events, filter, sortParam }).map(event => (
+            <MediaCard key={event.id} event={event} />
+          ))}
```

Comment:
why is `filterEvents()` called the second time here? I think using `filteredEvents` returned the first time would be good since this gets updated on every `setState` too.

Reply:
my badd 😂, will fix soon